keycloak/keycloak-admin-ui
# Realm settings(Client policies -> conditions): Update client-roles condition to match new design

## Motivation

https://marvelapp.com/prototype/2b3ghc06/screen/83236333

## Brief Description

Created a new `MultivaluedRoleComponent` to support selection of multiple client roles as per the updated design (link above). 

Client roles are now listed in a typeaheadMulti select variant. 

The work done in this PR also allows for the `client-roles` condition type to be saved and edited within an existing client policy. 

## Verification Steps


1. Go to Realm settings(Client policies -> profiles -> conditions).
2. Create a new client policy. 
3. Create a condition within the client policy.
4. Select the `client-roles` condition type.
5. Verify that the new multi-select component renders correctly.
6. Select from any of the client roles listed and verify that the data is saved correctly.
7. Click on the `client-roles` condition under "Conditions"
8. Edit the condition by selecting/deselecting client-roles from the dropdown.
9. Go back to the edited condition and verify it was saved successfully.  

## Checklist:


## Additional Notes


1. With the excellent work done in @agagancarczyk 's PRs for client policies, there have already been a couple times that this path has been changed and merged, which breaks client policies. https://github.com/keycloak/keycloak-admin-ui/pull/1499/files#diff-e9bd4be211c473f1940cf2eb43b009ca1a216d07632c521f494d6a2c4dba6171L12

   **It is important that we keep `/edit-profiles` in this path here, otherwise this breaks the client policy form.**
